asserts/store_asserts.go

```diff
@@ -0,0 +1,94 @@
+package asserts
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// EnterpriseStore holds an enterprise-store assertion, defining the
+// configuration needed to connect a device to the enterprise store.
+type EnterpriseStore struct {
```

Comment:
Can we adopt the name "CustomStore" and "custom-store" for this assertion?  This splits out the idea of a a store endpoint that the system may trust from the idea of why that store exists at all. The former is just a mechanism, while the latter is a specific use case.